### ISSUE TITLE
Missing `networkInactivityEnabled` on `appgatesdp_saml_identity_provider`

### ISSUE THREAD

**Issue by timvisher** [appgate/terraform-provider-appgatesdp#274]
Not filling in most of the Issue Template because this is just a missing attribute in the provider.

The 6.1 Release contains a new Inactivity Timeout feature which takes into account whether there's actually traffic flowing over the tunnel in addition to OS Input Activity.

Unfortunately, this can't currently be managed via `terraform` because a corresponding attribute isn't found in the provider. Can this be added and released?

Thanks in advance!

**Comment by timvisher** [comment]
👋 @danijeel. :)

Thanks again for the quick turnaround on this.

I'm using 1.0.4 against a 6.1 Collective and I just got the following:

```
2023-02-28T13:30:43.135-0500 [DEBUG] [aws-sdk-go] {}
╷
│ Error: network_inactivity_timeout_enabled is only available in 6.1 or higher
│
│   with module.identity_providers.appgatesdp_saml_identity_provider.google,
│   on identity_providers/google.tf line 1, in resource "appgatesdp_saml_identity_provider" "google":
│    1: resource "appgatesdp_saml_identity_provider" "google" {
│
╵
```

Any ideas what would cause that?

Looking at the logs on my Collective I don't even see an attempt to POST to `/admin/identity-providers` so it seems like the Provider itself is concluding that it shouldn't POST?

```
$ journalctl --since -1d -g 'POST /admin/identity-providers'
-- Logs begin at Mon 2023-02-20 20:52:44 UTC, end at Tue 2023-02-28 18:49:16 UTC. --
-- No entries --
```

On another Collective running an identical version of both the Provider (`1.0.4`) and Appgate (`6.1.0-31746-release`) I see the success POST going through…

Any help would be greatly appreciated. :)

**Comment by timvisher** [comment]
Also note that removing the attributes `network_inactivity_timeout_enabled` and `inactivity_timeout_minutes` from the config results in a successful POST against the misbehaving Collective:

```
… [28/Feb/2023:18:55:06 +0000] "POST /admin/identity-providers …
```

**Comment by danijeel** [comment]
Can you see what [client_version ](https://registry.terraform.io/providers/appgate/appgatesdp/latest/docs#client_version) you are using, either in the provider appgatesdp block or in your statefile? it should be `18` for version 6.1
```hcl
provider "appgatesdp" {
  client_version = XX// ??
}
```

or test  with environment variable, for example:
```
APPGATE_CLIENT_VERSION=18 terraform apply -auto-approve
```

**Comment by timvisher** [comment]
The provider block makes no mention of `client_version` (anywhere in my config).

Setting `APPGATE_CLIENT_VERSION=18` during plan and apply does seem to have fixed things. But I haven't needed to set that anywhere else in my config.

**Comment by timvisher** [comment]
One thing I noticed is that if I successfully create the IDP I can then _modify_ it to include the inactivity settings with no change (including setting client version).

**Comment by danijeel** [comment]
> Shouldn't `client_version` default to 18 in 1.0.4 since that adds `network_inactivity_timeout_enabled` and it seems to be required?

yes, You are right. We have forgotten to update it, we will fix it.

**Comment by timvisher** [comment]
> Shouldn't `client_version` default to 18 in 1.0.4 since that adds `network_inactivity_timeout_enabled` and it seems to be required?

@danijeel I just noticed the following in my Controller Logs.

> Warning from Controller
> 4/24/2023, 5:20:51 PM 21 occurrences
> Delete Message
> API version 'application/vnd.appgate.peer-v13+json' is still in use by "Terraform/1.3.9 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 appgatesdp/1.2.0 - 71.175.13.57" (current version is 'application/vnd.appgate.peer-v18+json'). Please update any external scripts as support for this version will be removed.

I guess that means that this was never fixed up through provider version 1.2.0?

**Comment by danijeel** [comment]
It's most likely because the provider config is missing [client version](https://registry.terraform.io/providers/appgate/appgatesdp/latest/docs#client_version) and it's computed automatically. This error message is [harmless](https://registry.terraform.io/providers/appgate/appgatesdp/latest/docs/guides/faq#controller-warning-api-version-deprecated-message) and will be generated each time you run terraform apply|refresh|destroy. If you want to avoid this error message, add `client_version` to your provider configuration or set the environment variable `APPGATE_CLIENT_VERSION`.
